### PR TITLE
CodyGarciaa/Legal-Rights-Prea-Pages

### LINKED ISSUE #7: Create PREA pages
## Todo: ##
- Create a button on the `LegalRights` page that leads to another page, this page will render buttons of all the videos 
- Clicking a button should lead to a new page, look into different video players for react-native so we can render a video some how.
   - import some sort of dummy video (maybe a rick roll lol), create a new folder called `assets` and a subfolder called `videos`. Try to render this video using whatever react-native video player you can find.

image
image

## Useful Links: ##
https://drawsql.app/teams/blueprint-5/diagrams/jdi-schema

https://supabase.com/dashboard/project/uhlkipxwhchenpwhbfvi/editor/30254?schema=public

### PR DESCRIPTION
**Pull request by CodyGarciaa**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- created the skeleton for the legal rights page and its prea module, not connected to any backend supabase though
- still using old react navigation, couldnt get nested react navigation to work well
- video is just looping without user interactions (video player) or sound i think

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

LegalRightsEnglish

LegalRightsSpanish

ModuleVideoPage

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- all the files i worked on is mostly just the LegalRights folder; few lines added to App and added assets folder with videos and images
- this should be reviewed all at once (latest commit)
- dont think theres any edge cases yet

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @philipye314

### CODE REVIEW

File: src/screens/LegalRights/index.tsx [hunk 1]
```diff
@@ -1,5 +1,163 @@
-import { Text } from 'react-native';
+import { useState } from 'react';
+import { Button, Image, Pressable, ScrollView, Text, View } from 'react-native';
+import placeholderPoster from '@/assets/images/placeholder.png';
+import { styles } from './styles';
 
-export default function LegalRights() {
-  return <Text>LEGAL RIGHTS!!</Text>;
+export default function LegalRights({ navigation }: { navigation: any }) {
+  function rick_rolls() {
```
Comment (philipye314) [inline src/screens/LegalRights/index.tsx]: just for consistency, let's make these into anonymous functions! Something like `const rickRolls = () => {}`. Also just be sure to use camelCase, as it's convention

File: src/screens/LegalRights/VideoPage/styles.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  captionButtons: {
+    borderRadius: 10,
+    width: 180,
+    height: 50,
+    borderWidth: 1,
+    backgroundColor: '#D9D9D9',
+    borderColor: '#D9D9D9',
+    shadowColor: 'black',
+    shadowOpacity: 0.05,
```
Comment (philipye314) [inline src/screens/LegalRights/VideoPage/styles.ts]: not super sure, but pretty sure these 2 lines are unnecessary for now since we don't have any shadow

File: src/screens/LegalRights/VideoPage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { Video } from 'expo-av';
+import { styles } from './styles';
+
+export default function VideoPage() {
+  return (
+    <ScrollView style={styles.container}>
+      <Video
+        source={require('@/assets/videos/da_link.mp4')}
+        rate={1.0}
+        volume={10.0}
```
Comment (philipye314) [inline src/screens/LegalRights/VideoPage/index.tsx]: nitpick but just leave this at 1, in the docs it says volume is an integer between 0.0 and 1.0. So just set it to 1 for now

File: src/screens/LegalRights/index.tsx [hunk 4]
```diff
@@ -1,5 +1,163 @@
-import { Text } from 'react-native';
+import { useState } from 'react';
+import { Button, Image, Pressable, ScrollView, Text, View } from 'react-native';
```
Comment (philipye314) [inline src/screens/LegalRights/index.tsx]: don't forget to delete this button import - it's not being used right now

**Review by philipye314 — COMMENTED** [review]
Looks mostly good to me! Couple of minor changes to take note of including standard naming conventions - just make sure to follow these going forward. Great work Cody

**Review by philipye314 — APPROVED** [review]
looks nice
